### PR TITLE
WIP: Fix unnamed structs appearing as numbers ##types

### PR DESCRIPTION
**Pull request by radare**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
Can you add a test please? Otherwise LGTM

**Comment by radare** [comment]
its quite red don't merge it

**Comment by radare** [comment]
wtf its running the wrong commit in some CI tasks?

File: shlr/tcc/tccgen.c [hunk 1]
```diff
@@ -1175,16 +1178,13 @@ static void struct_decl(CType *type, int u) {
...
-								ctype, name, varstr);
-							tcc_appendf ("%s.%s.%s.meta=%d\n",
-								ctype, name, varstr, type_bt);
+							tcc_appendf ("[+]%s.%s=%s\n", ctype, name, varstr);
+							tcc_appendf ("%s.%s.%s.meta=%d\n", ctype, name, varstr, type_bt);
 							/* compact form */
-							tcc_appendf ("%s.%s.%s=%s,%d,%d\n",
-								ctype, name, varstr, b, offset, arraysize);
+							tcc_appendf ("%s.%s.%s=%s,%d,%d\n", ctype, name, varstr, b, offset, arraysize);
 #if 0
```
Comment (thestr4ng3r) [inline shlr/tcc/tccgen.c:1188]: Test

File: shlr/tcc/tccgen.c [hunk 2]
```diff
@@ -1175,16 +1178,13 @@ static void struct_decl(CType *type, int u) {
...
 						char *varstr = get_tok_str (v, NULL);
 						type_to_str (b, sizeof(b), &type1, NULL);
-						{
+						if (is_named) {
 							int type_bt = type1.t & VT_BTYPE;
 							//eprintf("2: %s.%s = %s\n", ctype, name, varstr);
-							tcc_appendf ("[+]%s.%s=%s\n",
-								ctype, name, varstr);
-							tcc_appendf ("%s.%s.%s.meta=%d\n",
-								ctype, name, varstr, type_bt);
```
Comment (thestr4ng3r) [inline shlr/tcc/tccgen.c:1184]: Fcvjv

**Pull request closed without merging**